packages/compass-saved-aggregations-queries/src/hooks/use-grid-filters.tsx

```diff
@@ -0,0 +1,278 @@
+import React, { useState, useMemo, useRef, useEffect } from 'react';
+import { useId } from '@react-aria/utils';
+import {
+  css,
+  spacing,
+  Select,
+  Option,
+  cx,
+  TextInput,
+} from '@mongodb-js/compass-components';
+
+import type { Item } from '../stores/aggregations-queries-items';
+
+interface Tree {
+  value: string;
+  items?: Tree[];
+}
+
+interface SelectState {
+  database?: string;
+  collection?: string;
+}
+
+const selectContainer = css({
+  display: 'flex',
+  flexDirection: 'row',
+});
+
+const selectStyles = css({
+  marginRight: spacing[2],
+});
+
+const filterStyles = css({
+  display: 'flex',
+});
+
+const searchInputStyles = css({
+  width: '300px',
+  marginRight: spacing[2],
+});
+
+const RenderFilterSelect = ({
+  items,
+  onSelect,
+  defaultValue,
+  placeHolder = 'Select',
+}: {
+  items: Pick<Tree, 'value'>[];
+  onSelect: (value: string) => void;
+  defaultValue?: string;
+  placeHolder?: string;
+}) => {
+  const labelId = useId();
+  const controlId = useId();
+  const longestLabel = Math.max(
+    placeHolder.length,
+    ...items.map((item) => item.value.length)
+  );
+  return (
+    <Select
+      disabled={items.length === 0}
+      id={controlId}
+      aria-labelledby={labelId}
+      allowDeselect={true}
+      placeholder={placeHolder}
+      className={cx(
+        selectStyles,
+        css({ minWidth: `calc(${longestLabel}ch + ${spacing[6]}px)` })
+      )}
+      onChange={onSelect}
+      defaultValue={defaultValue}
+    >
+      {items.map((item) => (
+        <Option key={item.value} value={item.value}>
+          {item.value}
+        </Option>
+      ))}
+    </Select>
+  );
+};
+
+function useSearchFilter(): [React.ReactElement, string] {
+  const ref = useRef<HTMLInputElement>(null);
+  const [search, setSearch] = useState('');
+  useEffect(() => {
+    ref.current?.focus();
+  }, [search]);
+  const searchControls = useMemo(() => {
+    return (
+      <TextInput
+        ref={ref}
+        className={searchInputStyles}
+        aria-label="Search"
+        type="search"
+        placeholder="Search"
+        value={search}
+        defaultValue={search}
```

Comment:
Why are we passing both?
```suggestion
        value={search}
```